Configuration/AppUserStore.cs

```diff
@@ -0,0 +1,187 @@
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+
+namespace API.Configuration
+{
+    public class AppUserStore : IUserStore<Employee>, IUserPasswordStore<Employee>, IUserEmailStore<Employee>, IUserRoleStore<Employee>
+    {
+        private readonly ApiContext context;
+        private readonly ILogger logger;
+
+        public AppUserStore(ApiContext context, ILogger<AppUserStore> logger) {
+            this.context = context;
+            this.logger = logger;
+        }
+
+        public Task AddToRoleAsync(Employee user, string roleName, CancellationToken cancellationToken)
+        {
+            // WE can assing only single role :D 
+            // user.Department = DepartmentId.Admin;
+            // throw new System.NotImplementedException();
+            DepartmentId departmentId = (DepartmentId)Enum.Parse(typeof(DepartmentId), roleName, true);
+            user.Department = departmentId;
+            context.Employees.Update(user);
+            context.SaveChanges();
+            
+            return Task.FromResult(IdentityResult.Success);
+        }
+
+        public Task<IdentityResult> CreateAsync(Employee user, CancellationToken cancellationToken)
+        {
+            logger.LogInformation("Create user called: {}, {}", user.Username, user.Email);
+            context.Employees.Add(new Employee{
+                Username = user.Username,
+                Email = user.Email,
+                RegisterDate = new System.DateTime(), /// CIA
+                Password = user.Password,
+                FirstName = user.FirstName,
+                LastName = user.LastName,
+                PersonalCode = user.PersonalCode,
+                Department = user.Department,
+                Status = user.Status
+            });
+            context.SaveChanges();
+            return Task.FromResult(IdentityResult.Success);
+        }
+
+        public Task<IdentityResult> DeleteAsync(Employee user, CancellationToken cancellationToken)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public void Dispose()
+        {
+
+        }
+
+        public Task<Employee> FindByEmailAsync(string normalizedEmail, CancellationToken cancellationToken)
+        {
+            return Task.FromResult(context.Employees.Where(a => a.Email == normalizedEmail).FirstOrDefault<Employee>());
+        }
+
+        public Task<Employee> FindByIdAsync(string userId, CancellationToken cancellationToken)
+        {
+            return Task.FromResult(context.Employees.Where(a => a.Id.ToString() == userId).FirstOrDefault<Employee>());
+        }
+
+        public Task<Employee> FindByNameAsync(string normalizedUserName, CancellationToken cancellationToken)
+        {
+            logger.LogWarning("FindByNameAsync: {0} -> {}", normalizedUserName, context.Employees.ToList());
+            return Task.FromResult(context.Employees.Where(a => a.Username == normalizedUserName).FirstOrDefault<Employee>());
```

Comment:
Maybe we could simplify this LINQ to `context.Employees.FirstOrDefault(a => a.Username == normalizedUserName)`?